File: trunk/example/vb/helloworld/helloworld.vb
Imports System
Imports System.Runtime.InteropServices
Imports BotPlatformSDKLib

Module helloworld

    Sub MessageReceived(ByVal session As IRobotSession, ByVal message As IRobotMessage)

    End Sub

    <MTAThread()> Sub Main()
        Dim robotServerFactory As RobotServerFactory = New RobotServerFactory
        robotServerFactory.Init(2)

        Dim robotServer As RobotServer = robotServerFactory.CreateRobotServer("192.168.1.174", 6602)
        robotServer = robotServerFactory.CreateRobotServer("bottest.com", 6602)
        AddHandler robotServer.MessageReceived, AddressOf MessageReceived
        robotServer.Login("SP000125", "test111", 60000)

        Dim cmd As String = ""
        Do While cmd <> "exit"
            cmd = Console.ReadLine()
        Loop

        robotServer.Logout()
        robotServerFactory.Destroy()
    End Sub

End Module

File: trunk/example/cpp/helloworld/helloworld.cpp
#include "stdafx.h"
#include <atlbase.h>
#include <atlcom.h>
#include <string>
#include <iostream>

#import "../../../com/Bin/BotPlatformSDK.dll" named_guids no_namespace


_ATL_FUNC_INFO  s_info_onMessageReceived = { CC_STDCALL, VT_EMPTY, 2, {  VT_UNKNOWN, VT_UNKNOWN } };

class RobotServerEventsImpl : public IDispEventSimpleImpl<1, RobotServerEventsImpl, &DIID__IRobotServerEvents>
{
public:
    BEGIN_SINK_MAP(RobotServerEventsImpl)
        SINK_ENTRY_INFO(1, DIID__IRobotServerEvents, 3, &RobotServerEventsImpl::onMessageReceived, &s_info_onMessageReceived)
    END_SINK_MAP()

    HRESULT __stdcall onMessageReceived( IRobotSession* session, IRobotMessage* message )
    {
        session->SendText("Hello World!");
        std::cout << message->Text << std::endl;
        return S_OK;
    }
};

int _tmain(int argc, _TCHAR* argv[])
{
    CoInitializeEx(NULL, COINIT_MULTITHREADED);

    {
        CComPtr<IRobotServerFactory> spRobotServerFactory;
        spRobotServerFactory.CoCreateInstance( CLSID_RobotServerFactory, NULL, CLSCTX_INPROC );
        spRobotServerFactory->Init( 2 );

        IRobotServerPtr spRobotServer = spRobotServerFactory->CreateRobotServer( "server.botplatform.com", 6602 );
        RobotServerEventsImpl eventImpl;
        eventImpl.DispEventAdvise( spRobotServer );
        spRobotServer->Login( "YourSP", "Password", 60000 );

        std::string cmd;
        while ( true )
        {
            std::cin >> cmd;
            if ( cmd == "exit" )
                break;
        }

        spRobotServer->Logout();
        spRobotServerFactory->Destroy();
    }

    CoUninitialize();
	return 0;
}

